bot/exts/help_channels/_stats.py

```diff
@@ -22,21 +22,22 @@ def report_counts() -> None:
             log.warning(f"Couldn't find category {name!r} to track channel count stats.")
 
 
-async def report_complete_session(channel_id: int, is_auto: bool) -> None:
+async def report_complete_session(channel_id: int, closed_on: str) -> None:
     """
     Report stats for a completed help session channel `channel_id`.
 
     Set `is_auto` to True if the channel was automatically closed or False if manually closed.
     """
-    caller = "auto" if is_auto else "command"
-    bot.instance.stats.incr(f"help.dormant_calls.{caller}")
+    if closed_on != "command":
+        closed_on = f"auto.{closed_on}"
+    bot.instance.stats.incr(f"help.dormant_calls.{closed_on}")
 
     in_use_time = await _channel.get_in_use_time(channel_id)
     if in_use_time:
         bot.instance.stats.timing("help.in_use_time", in_use_time)
 
-    unanswered = await _caches.unanswered.get(channel_id)
-    if unanswered:
+    non_claimant_last_message_time = await _caches.non_claimant_last_message_times.get(channel_id)
+    if non_claimant_last_message_time is None:
         bot.instance.stats.incr("help.sessions.unanswered")
-    elif unanswered is not None:
+    else:
         bot.instance.stats.incr("help.sessions.answered")
```

Comment:
This is now broken due to
```py
        # non_claimant needs to be set too, to satisfy the condition in `_channel.get_closing_time` the first time.
        # Otherwise it will fall back to the old method if no other messages are sent.
        await _caches.non_claimant_last_message_times.set(message.channel.id, message.created_at)
```
The value should be None when the channel is claimed, but then you'd need to fix the fallback logic.

Reply:
Good point, I'll look into this 1